### ISSUE TITLE
HTTPS proxy not used when updating crate registry index

### ISSUE THREAD

**Issue by tjodden** [RustSec/rustsec-crate#146]
It seems like the HTTPS proxy is not used when updating the crate registry index, which is happens when checking for yanked crates. At work where using the HTTPS proxy is required for outgoing connections, this causes the 'Scanning' step to take a very long time. Running `cargo audit` through `strace` showed a connection to github.com timing out after 60 seconds or so. 

No error is reported, even though updating the crate index fails (and perhaps checking for yanked crates is therefore not done at all). I'd expect at least a warning that updating the index failed, and that it couldn't check for yanked crates (if that's the case).

It would also be nice with a status message when the index is being updated, as this is a network operation which may take a long time on slow networks even when it does work.

**Comment by tarcieri** [comment]
I'm guessing most of this will require upstream fixes in the [`crates-index` crate](https://github.com/frewsxcv/rust-crates-index) similar to the one you already did for the `rustsec` crate (#126), and perhaps a more expedient option would be a command-line option and/or configuration setting to disable automatically updating the crate registry index.

> No error is reported, even though updating the crate index fails (and perhaps checking for yanked crates is therefore not done at all). I'd expect at least a warning that updating the index failed, and that it couldn't check for yanked crates (if that's the case).

That's strange. The code in the `rustsec` crate itself actually doing the update is one line and is theoretically propagating any errors:

https://github.com/RustSec/rustsec-crate/blob/bd7a348/src/registry.rs#L17

> It would also be nice with a status message when the index is being updated, as this is a network operation which may take a long time on slow networks even when it does work.

That's easy enough to add.

**Comment by tjodden** [comment]
> I'm guessing most of this will require upstream fixes in the [`crates-index` crate](https://github.com/frewsxcv/rust-crates-index) similar to the one you already did for the `rustsec` crate (#126), 

I'll have a look at the `crates-index` crate.

> > No error is reported, even though updating the crate index fails (and perhaps checking for yanked crates is therefore not done at all). I'd expect at least a warning that updating the index failed, and that it couldn't check for yanked crates (if that's the case).
> 
> That's strange. The code in the `rustsec` crate itself actually doing the update is one line and is theoretically propagating any errors:
> 
> https://github.com/RustSec/rustsec-crate/blob/bd7a348/src/registry.rs#L17

Yes, however the error is later ignored:

```rust
if let Ok(mut yanked) = find_yanked_crates(lockfile) {
    warnings.append(&mut yanked);
}
```

https://github.com/RustSec/rustsec-crate/blob/35a939/src/report.rs#L55

**Comment by tarcieri** [comment]
I believe this PR should improve the diagnostics for interactions with the crates.io index. It also adds some configuration options to disable automatically updating the index (or disable yanked crate auditing entirely) if that's helpful:

https://github.com/RustSec/cargo-audit/pull/200/files

**Comment by tjodden** [comment]
I've opened a PR against the `crates-index` crate:
frewsxcv/rust-crates-index#31

**Comment by tarcieri** [comment]
I think this was fixed in `crates-index` v0.14, but it looks like they bumped MSRV to 1.40 in that release as well (see #151).

This crate doesn't have an official MSRV policy, but I prefer not to bump MSRV in a point release so I guess we'll upgrade with the next minor release.

**Comment by tarcieri** [comment]
We're now using `crates-index` v0.15, so I'll consider this addressed.
